ambry-account/src/main/java/com/github/ambry/account/JsonAccountService.java

```diff
@@ -0,0 +1,291 @@
+    // We'll only be iterating over the result in order so we use a linked list.
+    Set<Account> accounts = new HashSet<>();
+
+    for (int i = 0; i < accountArray.length(); i++) {
+      JSONObject accountJson = accountArray.getJSONObject(i);
+      accounts.add(Account.fromJson(accountJson));
+    }
+
+    if (hasDuplicateAccountIdOrName(accounts)) {
+      throw new IllegalArgumentException("Duplicate id or name exists in the accountsById to update");
+    }
+
+    return accounts;
+  }
+
+  private void notifyAccountUpdateConsumers(Set<Account> updatedAccounts) {
+    for (Consumer<Collection<Account>> accountUpdateConsumer : accountUpdateConsumers) {
+      long startTime = System.currentTimeMillis();
+      try {
+        accountUpdateConsumer.accept(updatedAccounts);
+        long consumerExecutionTimeInMs = System.currentTimeMillis() - startTime;
+
+        logger.trace("Consumer={} has been notified for account change, took {} ms", accountUpdateConsumer,
+            consumerExecutionTimeInMs);
+        accountServiceMetrics.accountUpdateConsumerTimeInMs.update(consumerExecutionTimeInMs);
+      } catch (Exception e) {
+        logger.error("Exception occurred when notifying accountUpdateConsumer={}", accountUpdateConsumer, e);
+      }
+    }
+  }
+
+  /** How often we check if the JSON account file has been modified in milliseconds. */
+  private static int fileWwatchIntervalInMs = 10 * 1000;
+  private Thread watchAccountFileThread;
+
+  /**
+   * Starts a thread which checks (at fixed intervals) if the JSON account file has changed. If it has it triggers
+   * and update of the accounts.
+   */
+  private void watchJsonFile() {
+    watchAccountFileThread = new Thread(() -> {
+      logger.info("Starting JSON account file monitor. File check interval is {}ms", fileWwatchIntervalInMs);
+
+      FileTime accountFilePreviousModified = null;
+      long startTimeMs = -1;
+      boolean success = false;
+
+      while (!Thread.interrupted()) {
+        try {
+          startTimeMs = System.currentTimeMillis();
+          success = false;
+
+          logger.trace("Checking if JSON account file has changed.");
+
+          if (!Files.exists(accountFile)) {
+            logger.warn("Unable to open JSON account file '{}'", accountFile.toString());
+            continue;
+          }
+
+          FileTime accountFileLastModified = Files.getLastModifiedTime(accountFile);
+          if (accountFilePreviousModified == null
+              || accountFilePreviousModified.compareTo(accountFileLastModified) < 0) {
+            logger.debug("JSON Account file has changed or has never been loaded before. "
+                    + "Previous last know modified time: {}, new last known modified time: {}", accountFilePreviousModified,
+                accountFileLastModified);
+
+            updateAccountsFromFile();
+            accountFilePreviousModified = accountFileLastModified;
+          }
+
+          success = true;
```

Comment:
I didn't really like the use of this variable either (instead doing it in the `catch`, etc.) but I thought it was convention (I modeled this after the `HelixAccountService`). However looking at it again I realize now I misinterperted it. It's simply because the `helixStore.update` returns a boolean. I've modified it, let me know if this is more in line with your expectations!